[Grid][Overflow] Start considering grid areas in layout overflow.

sammygill opened the pull request:
#### bf6d6e8f80af5c5c99fb3481e55903be90ab688a
<pre>
[Grid][Overflow] Start considering grid areas in layout overflow.
<a href="https://bugs.webkit.org/show_bug.cgi?id=260292">https://bugs.webkit.org/show_bug.cgi?id=260292</a>
<a href="https://rdar.apple.com/113985286">rdar://113985286</a>

Reviewed by Alan Baradlay.

The grid area, which is composed of the rows and columns, can
potentially overflow the grid container itself even if the grid does
not have any grid items or none of the grid items overflow the grid.

We can let RenderGrid override computeOverflow in order to determine
if the grid area is overflowing in either dimension. This is done by
consulting the position of the last column and row line, which are
offsets within the border box of the grid, to determine the bounds of
the grid area, and passing it off to addLayoutOverflow to potentially
expand the bounds of the overflow rect.

* LayoutTests/imported/w3c/web-platform-
* Source/WebCore/rendering/RenderGrid.cp
(WebCore::RenderGrid::computeOverflow):# Please populate the above commit message. Lines starting
* Source/WebCore/rendering/RenderGrid.h:

Canonical link: <a href="https://commits.webkit.org/281594@main">https://commits.webkit.org/281594@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/504d5903a30bb2b4b0ac95d08ae9a001ff55927a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60402 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39753 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12960 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/64321 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10933 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62532 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47425 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11166 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/64321 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7613 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62433 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37040 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52330 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/64321 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33738 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9556 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/9850 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/55618 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9844 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66053 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4335 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/9674 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4354 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52305 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/56424 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13422 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3610 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/35563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/36645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/37734 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36389 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderGrid.cpp:
```diff
@@ -2630,4 +2630,18 @@ void RenderGrid::GridWrapper::resetCurrentGrid() const
+    // FIXME: We should handle RTL and other writing modes also.
+    if (style().direction() == TextDirection::LTR && isHorizontalWritingMode()) {
+        auto gridAreaSize = LayoutSize { m_columnPositions.last(), m_rowPositions.last() };
+        addLayoutOverflow({ LayoutPoint(), gridAreaSize });
```
nit, this could be just `addLayoutOverflow({ { }, gridAreaSize });` but it is really okay to land as is.

sammygill committed:
[Grid][Overflow] Start considering grid areas in layout overflow.
https://bugs.webkit.org/show_bug.cgi?id=260292
rdar://113985286

Reviewed by Alan Baradlay.

The grid area, which is composed of the rows and columns, can
potentially overflow the grid container itself even if the grid does
not have any grid items or none of the grid items overflow the grid.

We can let RenderGrid override computeOverflow in order to determine
if the grid area is overflowing in either dimension. This is done by
consulting the position of the last column and row line, which are
offsets within the border box of the grid, to determine the bounds of
the grid area, and passing it off to addLayoutOverflow to potentially
expand the bounds of the overflow rect.

* LayoutTests/imported/w3c/web-platform-
* Source/WebCore/rendering/RenderGrid.cp
(WebCore::RenderGrid::computeOverflow):# Please populate the above commit message. Lines starting
* Source/WebCore/rendering/RenderGrid.h:

Canonical link: https://commits.webkit.org/281594@main
